### ISSUE TITLE
R04 Disable GraphQL introspection in deployed environments

### ISSUE THREAD

**Issue by rikukissa** [opencrvs/opencrvs-core#7501]
### Security assessment details

**Technical Overview** 

>The introspection mechanism allowed any logged-in user to request information about the GraphQL 
schema. This allows an attacker to gain direct insight into available queries, mutations and 
subscriptions. 

**Potential Impact if Exploited** 
>Viewing the complete GraphQL schema can aid the attacker in creating more targeted attacks 
against the application. 

**Recommendations** 
>Ensure that GraphQL introspection is disabled within the application. 
>To disable introspection, set the value of the introspection configuration key on the Apollo Server 
instance like so: 

```ts
const server = new ApolloServer({ 
  typeDefs, 
  resolvers, 
  introspection: process.env.NODE_ENV !== 'production' 
}); 
```
>For further information, refer to the following post on the Apollo GraphQL blog: 
https://www.apollographql.com/blog/why-you-should-disable-graphql-introspection-in-production 
Instance

**Comment by makelicious** [comment]
@rikukissa is events still the relevant parent for this?

**Comment by rikukissa** [comment]
I don't think it is. Let's move it outside. It will most likely then target 1.8 or 1.7.1

**Comment by rikukissa** [comment]
Not relevant in v2.0
